### PR TITLE
Use SchemaRegistryClient to register schemas in Product tests

### PR DESCRIPTION
**Pull request by Praveen2112**

### CODE REVIEW

File: testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/singlenode-kafka/tempto-configuration.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+databases:
```
Comment (kokosing) [inline testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/singlenode-kafka/tempto-configuration.yaml:1]: It is not a database. Maybe just:
```
schema-registry:
      url: http://schema-registry:8081
```

File: testing/trino-product-tests/pom.xml [hunk 2]
```diff
@@ -157,6 +157,12 @@
             <scope>runtime</scope>
         </dependency>
 
+        <dependency>
+            <groupId>javax.ws.rs</groupId>
+            <artifactId>javax.ws.rs-api</artifactId>
+            <scope>runtime</scope>
```
Comment (kokosing) [inline testing/trino-product-tests/pom.xml:163]: Why is this needed? Does it need a comment in the code?

File: testing/trino-product-tests/src/main/java/io/trino/tests/kafka/TestKafkaAvroReadsSmokeTest.java [hunk 3]
```diff
@@ -229,10 +230,6 @@ public String toString()
         }
     }
 
-    /**
-     * Using CLI and custom serializer as Confluent dependencies conflicts with Hive dependencies
```
Comment (kokosing) [inline testing/trino-product-tests/src/main/java/io/trino/tests/kafka/TestKafkaAvroReadsSmokeTest.java:233]: You are still using custom serializer.

File: testing/trino-product-tests/src/main/java/io/trino/tests/utils/SchemaRegistryClientUtils.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.trino.tempto.configuration.Configuration;
+
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.trino.tempto.context.ThreadLocalTestContextHolder.testContext;
+
+public class SchemaRegistryClientUtils
```
Comment (kokosing) [inline testing/trino-product-tests/src/main/java/io/trino/tests/utils/SchemaRegistryClientUtils.java:25]: final

File: testing/trino-product-tests/src/main/java/io/trino/tests/utils/SchemaRegistryClientUtils.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+{
+    private SchemaRegistryClientUtils()
```
Comment (kokosing) [inline testing/trino-product-tests/src/main/java/io/trino/tests/utils/SchemaRegistryClientUtils.java:27]: private SchemaRegistryClientUtils() {}

File: testing/trino-product-tests/src/main/java/io/trino/tests/kafka/TestKafkaAvroReadsSmokeTest.java [hunk 6]
```diff
@@ -242,8 +239,10 @@ public String toString()
         {
             try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
                 out.write((byte) 0);
-                SchemaId schemaId = registerSchema(topic, schema);
-                out.write(Ints.toByteArray(schemaId.getId()));
+
+                out.write(Ints.toByteArray(getSchemaRegistryClient().register(
```
Comment (kokosing) [inline testing/trino-product-tests/src/main/java/io/trino/tests/kafka/TestKafkaAvroReadsSmokeTest.java:243]: extract variable `int schemaId`, that way this line will be less dense

File: testing/trino-product-tests/src/main/java/io/trino/tests/utils/SchemaRegistryClientUtils.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    {
+        Map<String, Object> schemaRegistryProperties = testContext().getDependency(Configuration.class)
+                .getSubconfiguration("databases.kafka.schema-registry").asMap();
+
+        checkArgument(schemaRegistryProperties.containsKey("url"), "schemaRegistry url is missing");
+
+        return new CachedSchemaRegistryClient(
+                (String) schemaRegistryProperties.get("url"),
+                100,
+                schemaRegistryProperties);
```
Comment (kokosing) [inline testing/trino-product-tests/src/main/java/io/trino/tests/utils/SchemaRegistryClientUtils.java:41]: do you need to filter `url`?

File: testing/trino-product-tests/src/main/java/io/trino/tests/utils/SchemaRegistryClientUtils.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    public static SchemaRegistryClient getSchemaRegistryClient()
+    {
+        Map<String, Object> schemaRegistryProperties = testContext().getDependency(Configuration.class)
+                .getSubconfiguration("databases.kafka.schema-registry").asMap();
+
+        checkArgument(schemaRegistryProperties.containsKey("url"), "schemaRegistry url is missing");
+
+        return new CachedSchemaRegistryClient(
+                (String) schemaRegistryProperties.get("url"),
+                100,
```
Comment (kokosing) [inline testing/trino-product-tests/src/main/java/io/trino/tests/utils/SchemaRegistryClientUtils.java:40]: what is 100? Maybe extract a variable that is named exactly as constructor parameter?

**Review by kokosing — COMMENTED** [review]
Nice, only minor comments. Mind automation.

**Comment by Praveen2112** [comment]
@kokosing AC

**Review by kokosing — APPROVED** [review]

**Review by skrzypo987 — APPROVED** [review]

**Review by lukasz-walkiewicz — APPROVED** [review]
